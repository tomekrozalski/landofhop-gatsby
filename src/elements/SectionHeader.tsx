import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  margin: 4rem 0 2rem 0;
  position: relative;

  ::before {
    display: block;
    width: 100%;
    height: 1px;
    content: '';
    background-color: var(--color-bright);
    position: absolute;
    top: 50%;
    left: 0;
    z-index: -1;
  }
`;

const Header = styled.h2`
  display: inline-block;
  margin: 0 4rem;
  padding: 0 1rem;
  font: var(--font-weight-regular) 1.5rem / 1 var(--font-primary);
  background-color: var(--color-white);
`;

const SectionHeader: React.FC = ({ children }) => (
  <Wrapper>
    <Header>{children}</Header>
  </Wrapper>
);

export default SectionHeader;
