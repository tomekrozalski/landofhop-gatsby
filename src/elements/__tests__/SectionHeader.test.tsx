import React from 'react';
import { render } from '@testing-library/react';

import SectionHeader from '../SectionHeader';

test('should render', () => {
  const { container, debug } = render(<SectionHeader>Bum!</SectionHeader>);

  debug();

  expect(container).toMatchSnapshot();
  expect(1 + 1).toBe(2);
});
