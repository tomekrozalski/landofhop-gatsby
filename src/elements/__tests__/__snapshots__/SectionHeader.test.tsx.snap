// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <div
    class="sc-AxjAm kFpWZV"
  >
    <h2
      class="sc-AxirZ dqRrbJ"
    >
      Bum!
    </h2>
  </div>
</div>
`;
